client/my-sites/stats/pages/realtime/chart.tsx

```diff
@@ -0,0 +1,109 @@
+import moment from 'moment';
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import AsyncLoad from 'calypso/components/async-load';
+import wpcom from 'calypso/lib/wp';
+import { getSiteOption } from 'calypso/state/sites/selectors';
+import { parseChartData } from 'calypso/state/stats/lists/utils';
+import PageLoading from '../shared/page-loading';
+
+type Unit = 'hour' | 'day' | 'week' | 'month' | 'year';
+
+interface QueryStatsVisitsParams {
+	unit: Unit;
+	date: string;
+	quantity: number;
+	stat_fields: string;
+}
+
+interface chartMinuteDataTypes {
+	[ key: string ]: number;
+}
+
+function queryStatsVisits( siteId: number, params: QueryStatsVisitsParams ) {
+	return wpcom.req.get( `/sites/${ siteId }/stats/visits`, params );
+}
+// TODO: Change to 60 for production.
+const UPDATE_INTERVAL_IN_SECONDS = 5;
+// TODO: Change to 30 after resolving the date format issue on the X-axis.
+const MINUTE_DATA_LENGTH = 5;
+
+const RealtimeChart = ( { siteId }: { siteId: number } ) => {
+	const gmtOffset = useSelector( ( state: object ) =>
+		getSiteOption( state, siteId, 'gmt_offset' )
+	) as number;
+	const [ viewsData, setViewsData ] = useState( {} as chartMinuteDataTypes );
+
+	useEffect( () => {
+		const intervalId = setInterval( () => {
+			// Index the chart data by YYYY-MM-DD HH:mm:00.
+			const adjustedDatetime = moment().utcOffset( gmtOffset ).format( 'YYYY-MM-DD HH:mm:00' );
```

Comment:
Are we able to reuse `momentSiteZone` here?

Reply:
I tried it, but the current hour might not be accurate inside the `setInterval` callback. I will test again if this approach works for minutes across the hour.